addons/website_quote/__init__.py

```diff
@@ -3,3 +3,14 @@
 
 from . import controllers
 from . import models
+
+from odoo.api import Environment, SUPERUSER_ID
+
+def _install_sale_payment(cr):
+    env = Environment(cr, SUPERUSER_ID, {})
+    sale_payment = env['ir.module.module'].search([
+        ('name', '=', 'sale_payment'),
+    ], limit=1)
+
+    if sale_payment.state == 'uninstalled':
+        sale_payment.state = 'to install'
```

Comment:
This is bad, if `sale_payment` has dependencies that have not been installed, you will install `sale_payment` but not its dependencies, the dependency prepping is done at the `button_install` level [0].

Even if `sale_payment` has no dependencies (i didn't check) it's still bad practice, module exclusivity and auto-install are done at the same level.

You probably didn't have luck with the `button_immediate_ïnstall` because it commits the current transaction and creates a brand new registry, try `button_install` instead.

[0] [button_install](https://github.com/odoo/odoo/blob/1d245deddd88893603383680b454822da24718aa/odoo/addons/base/module/module.py#L396)

Reply:
> Even if sale_payment has no dependencies (i didn't check) it's still bad practice, module exclusivity and auto-install are done at the same level.

Yeah, in this case the dependencies of `sale_payment` are a subset of those of `website_quote` which explains why I didn't run into issues, but I do agree it's not the right way to do it.

> You probably didn't have luck with the button_immediate_install because it commits the current transaction and creates a brand new registry, try button_install instead.

You are correct, I turned the hook into a `*post*_init_hook` and used `button_install` instead.